fuel-p2p/src/gossipsub/builder.rs

```diff
@@ -1,31 +1,29 @@
-use libp2p::{
-    gossipsub::{
-        FastMessageId,
-        Gossipsub,
-        GossipsubConfigBuilder,
-        GossipsubMessage,
-        MessageAuthenticity,
-        MessageId,
-        PeerScoreParams,
-        PeerScoreThresholds,
-        RawGossipsubMessage,
-    },
-    identity::Keypair,
+use libp2p::gossipsub::{
+    FastMessageId,
+    Gossipsub,
+    GossipsubConfigBuilder,
+    GossipsubMessage,
+    MessageAuthenticity,
+    MessageId,
+    PeerScoreParams,
+    PeerScoreThresholds,
+    RawGossipsubMessage,
 };
 use sha2::{
     Digest,
+    Sha224,
     Sha256,
 };
 
 use crate::config::P2PConfig;
 
-pub fn build_gossipsub(local_key: &Keypair, p2p_config: &P2PConfig) -> Gossipsub {
+pub fn build_gossipsub(p2p_config: &P2PConfig) -> Gossipsub {
     let gossip_message_id = move |message: &GossipsubMessage| {
         MessageId::from(&Sha256::digest(&message.data)[..20])
```

Comment:
We should hash the entire message for uniqueness, otherwise we may drop messages that are actually different from each other.